ui/pages/confirmation/util.ts

```diff
@@ -0,0 +1,155 @@
+import { ResultComponent } from '@metamask/approval-controller';
+
+type TemplateRendererComponent = {
+  key: string;
+  element: string;
+  props?: Record<string, unknown>;
+  children?:
+    | string
+    | TemplateRendererComponent
+    | (string | TemplateRendererComponent)[];
+};
+
+/**
+ * Processes an error message or ResultComponent and returns a TemplateRendererComponent
+ * or an array of strings | TemplateRendererComponents.
+ *
+ * @param input - The message or component to process.
+ * @param fallback - The fallback message to use when the input is not valid.
+ * @returns The processed error component.
+ */
+export function processError(
+  input: undefined | string | ResultComponent | ResultComponent[],
+  fallback: string,
+): TemplateRendererComponent | (string | TemplateRendererComponent)[] {
+  let currentInput = convertResultComponents(input);
+
+  if (!currentInput) {
+    currentInput = fallback;
+  }
```

Comment:
Couldn't this be replaced with `const currentInput = convertResultComponents(input) || fallback;`?